velox/type/Timestamp.cpp

```diff
@@ -213,47 +212,69 @@ bool epochToUtc(int64_t epoch, std::tm& tm) {
   return true;
 }
 
-std::string tmToString(
+std::string Timestamp::tmToString(
     const std::tm& tmValue,
     int nanos,
     const TimestampToStringOptions& options) {
   VELOX_DCHECK_GE(nanos, 0);
   VELOX_DCHECK_LT(nanos, 1'000'000'000);
-  int width = options.precision;
+  int width = static_cast<int>(options.precision);
   std::string out;
-  out.reserve(options.dateOnly ? 10 : 26 + width);
-  int n = kTmYearBase + tmValue.tm_year;
-  bool negative = n < 0;
-  if (negative) {
-    out += '-';
-    n = -n;
-  }
-  while (n > 0) {
-    out += '0' + n % 10;
-    n /= 10;
+
+  std::string::size_type capacity = -1;
```

Comment:
For readability, consider extracting logic for computing capacity into a helper method.